tests/cql_query_test.cc

```diff
@@ -4261,3 +4261,106 @@ SEASTAR_TEST_CASE(test_rf_expand) {
         });
     });
 }
+
+SEASTAR_TEST_CASE(test_describe_simple_schema) {
+    return do_with_cql_env_thread([] (cql_test_env& e) {
+        std::string create_normal_table = "CREATE  TABLE ks.cf (\n"
+                "    pk blob,\n"
+                "    col1 blob,\n"
+                "    col2 blob,\n"
+                "    PRIMARY KEY (pk)\n"
+                ") WITH bloom_filter_fp_chance = 0.01\n"
+                "    AND caching = {'keys': 'ALL','rows_per_partition': 'ALL'}\n"
+                "    AND comment = ''\n"
+                "    AND compaction = {'class': 'SizeTieredCompactionStrategy'}\n"
+                "    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}\n"
+                "    AND crc_check_chance = 1\n"
+                "    AND dclocal_read_repair_chance = 0.1\n"
+                "    AND default_time_to_live = 0\n"
+                "    AND gc_grace_seconds = 864000\n"
+                "    AND max_index_interval = 2048\n"
+                "    AND memtable_flush_period_in_ms = 0\n"
+                "    AND min_index_interval = 128\n"
+                "    AND read_repair_chance = 0\n"
+                "    AND speculative_retry = '99.0PERCENTILE';\n";
+        e.execute_cql(create_normal_table).get();
+        auto schema = e.local_db().find_schema("ks", "cf");
+        std::ostringstream ss;
+
+        schema->describe(ss);
+        BOOST_CHECK_EQUAL(ss.str(), create_normal_table);
+        e.execute_cql("DROP TABLE ks.cf").get();
+        e.execute_cql(ss.str()).get();
+
+        std::string create_table_with_compund_key = "CREATE  TABLE ks.cf1 (\n"
```

Comment:
There are other variants you should try, e.g., multiple partition key columns (which is a special case you have special code with parentheses for), and importantly - uppercase table names, quoted column names, etc.
To reduce the amount of duplicate code, you can have an array of strings with just the different part (appending at the end the common part with all those silly parameters), and do the test with a loop instead of duplicate code.

Reply:
sounds good, I'll squeeze it into a single test with a loop